src/Drips.sol

```diff
@@ -15,34 +22,32 @@ library Drips {
     /// Limits cost of changes in drips configuration.
     uint32 internal constant MAX_DRIPS_RECEIVERS = 100;
 
-    /// @notice Emitted when drips from a user to a receiver are updated.
-    /// Funds are being dripped on every second between the event block's timestamp (inclusively)
-    /// and`endTime` (exclusively) or until the timestamp of the next drips update (exclusively).
-    /// @param userId The dripping user ID.
-    /// @param receiver The receiver user ID
+    /// @notice Emitted when the drips configuration of a user is updated.
+    /// @param userId The user ID.
     /// @param assetId The used asset ID
-    /// @param amtPerSec The new amount per second dripped from the user
-    /// to the receiver or 0 if the drips are stopped
-    /// @param endTime The timestamp when dripping will stop,
-    /// always larger than the block timestamp or equal to it if the drips are stopped
-    event Dripping(
+    /// @param receiversHash The drips receivers list hash
+    /// @param balance The new drips balance. These funds will be dripped to the receivers.
+    event DripsSet(
         uint256 indexed userId,
-        uint256 indexed receiver,
         uint256 indexed assetId,
-        uint128 amtPerSec,
-        uint64 endTime
+        bytes32 indexed receiversHash,
+        uint128 balance
     );
 
-    /// @notice Emitted when the drips configuration of a user is updated.
+    /// @notice Emitted when a user is seen in a drips receivers list.
+    /// @param receiversHash The drips receivers list hash
     /// @param userId The user ID.
-    /// @param assetId The used asset ID
-    /// @param balance The new drips balance. These funds will be dripped to the receivers.
-    /// @param receivers The new list of the drips receivers.
-    event DripsUpdated(
+    /// @param amtPerSec The amount per second being dripped. Must never be zero.
+    /// @param start The timestamp when dripping should start.
+    /// If zero, use the timestamp when drips are configured.
+    /// @param duration The duration of dripping.
+    /// If zero, drip until balance runs out.
+    event DripsReceiverSeen(
```

Comment:
Why not just `DripsReceiver`?

Reply:
It would be nice, but won't it be confusing? Events usually describe things that happened, `DripsReceiver` doesn't happen. If you think it's ok, I'll be glad to shorten the name.

Reply:
We can keep it. No strong opinion